lib/src/markdown_content.dart

```diff
@@ -52,7 +60,8 @@ List<T> _unique<T>(Iterable<T> l) => l.toSet().toList();
 
 /// Scans a markdown file and extracts its content.
 Future<ExctractedMarkdownContent> scanMarkdownFileContent(File file) async {
-  final text = await file.readAsString();
+  final bytes = await file.readAsBytes();
+  final text = utf8.decode(bytes, allowMalformed: true);
```

Comment:
The 100% correct code would first try to decode it without `allowMalformed`, and if that fails, set the flag. However, I don't think that too many valid documents have the replacement character as part of their content.

Reply:
But isn't it cheap to do the right thing?

Reply:
Thinking about it: assuming that most files are well-formed, it may be cheaper to do a second pass for malformed ones than always doing a string.contains().